### PR TITLE
Milestone 2: Implement components

### PR DESCRIPTION
**Pull request by rukundoeric**
## What does this do? 
*  Implement components
*  Connect components to redux store
## Description of Task to be completed? 
*  In this milestone, I implemented components and connected some of them to redux store.
## How should this be manually tested? 
NOTE: Make sure you have Node.js and npm or yarn installed in your PC.
- Clone this repo: `git clone https://github.com/rukundoeric/bookstore.git`
- Navigate into project directory: `cd bookstore`
- Install all packages: `yarn install or npm install`
- Start the app: `yarn start or npm start`

### CODE REVIEW

**Review by krys2fa — CHANGES_REQUESTED** [review]
# Status: Marked as Invalid! ⚠️  
Please note that your Code Review Request is being marked as invalid. 
This is because the configs in .github/workflows/ must be exactly the same as that in the [Microverse linter config repo](https://github.com/microverseinc/linters-config/blob/master/react-redux/.github/workflows/linters.yml). 

💡 Do not make any changes in config files - they represent style guidelines that you share with your team - which is a group of all Microverse students.
If you think that the change is necessary - open a [Pull Request in this repository](https://github.com/microverseinc/linters-config/blob/master/README.md#contributing).


  

Friendly reminder: You should make sure that your repo looks professional before review.
Please take a step back and read a lesson here https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243642/.
Then apply 3 simple rules from this lesson (it will be easy as you can use a template provided there).
This Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready!

File: src/components/containers/BooksForm.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
```
Comment (bellom) [inline src/components/containers/BooksForm.js]: You are not modifying linter config files directly, but you are disabling ESLint rules with the inline Eslint comments, and that's kind of bypassing ESLint configuration which represent style guidelines that you share with your team - which is a group of all Microverse students. (screenshot from Pathwright).

 Please remove ESLint disabling rules comments and fix your code accordingly.
I'm sure this is not going to be an issue for you, you got this 💪.



Kindly apply to the rest of the eslint comment 👍 .

File: src/components/containers/BooksForm.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import React from 'react';
+import { connect } from 'react-redux';
```
Comment (bellom) [inline src/components/containers/BooksForm.js]: Kindly improve the project structure such that you have components for stateless components and a container/page folder for other components that use the state.

React bindings for Redux separate presentational components from container components. This approach can make your app easier to understand and allow you to more easily reuse components. In your components directory (presentational components) you have components that are not connected to the Redux store, and on your containers directory, you have components that are connected to the Redux store. [Read More...](https://redux.js.org/basics/usage-with-react#presentational-and-container-components.)

Screenshot from the pathwright
image

File: src/components/containers/BooksList.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
+/* eslint-disable react/forbid-prop-types */
+import React from 'react';
+import key from 'uniqid';
+import { connect } from 'react-redux';
```
Comment (bellom) [inline src/components/containers/BooksList.js:4]: Kindly improve the project structure such that you have components for stateless components and a container/page folder for other components that use the state.

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi @rukundoeric  :clap: 

### Awesome job so far!! :+1: @rukundoeric 

#### You have done wonderfully well, you passed `almost` all the checklist for this project :+1: just a few issues left to work on............ :slightly_smiling_face: 

# Status: Changes Required! ♻️ 

**Kindly check out my comments below;**

### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

**Review by uche-inyama — CHANGES_REQUESTED** [review]
## Changes Required

### Hi @rukundoeric :wave: 

We still have some changes to make. From the last review you got, the files in the components folder was meant to be separated into presentational and container components. Looking at your project folder, the separation has been done, but the naming isn't correct. Kindly change the items folder to **presentational**

#### Implement the change, then submit another form for a code review.

### Have a great day :slightly_smiling_face: 


#### ``` Uchechukwu Henry Inyama ```

**Review by uche-inyama — APPROVED** [review]
## Approved

### Hi @rukundoeric :wave: 

Move on with your next task. 

### Have a great day :slightly_smiling_face: 

#### ``` Uchechukwu Henry Inyama ```

**Pull request closed without merging**